File: src/main/java/org/pixeltime/enchantmentsenhance/commands/console/SubConsoleCommand.java
package org.pixeltime.enchantmentsenhance.commands.console;


import org.bukkit.command.CommandSender;

public abstract class SubConsoleCommand {

    /**
     * /<command> <subcommand> args[0] args[1]
     */
    public SubConsoleCommand() {
    }


    public abstract void onCommand(CommandSender sender, String[] args);


    public abstract String name();


    public abstract String info();


    public abstract String[] aliases();
}

File: modules/EnchantmentsEnhance-Plugin/src/main/kotlin/org/pixeltime/enchantmentsenhance/command/SubConsoleCommand.kt
package org.pixeltime.enchantmentsenhance.command


import org.bukkit.command.CommandSender
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

/**
 * /<command> <subcommand> args[0] args[1]
</subcommand></command> */
abstract class SubConsoleCommand {

    /**
     * Console Command
     *
     * @param sender
     * @param args
     */
    abstract fun onCommand(sender: CommandSender, args: Array<String>)

    /**
     * @return Command name.
     */
    abstract fun name(): String

    /**
     * @return Command information.
     */
    abstract fun usage(): String

    /**
     * @return Command aliases.
     */
    abstract fun aliases(): Array<String>

    fun info(): String {
        return "&b${usage()} ${SettingsManager.lang.getString("help.${name()}")}"
    }
}
